src/hooks/campaigns/useGetLeaderboardGrantProgram.ts

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios'
+import { stringify } from 'querystring'
+import useSWR, { mutate } from 'swr'
+
+import { SWR_KEYS } from 'constants/index'
+import { ProjectRanking } from 'types/grantProgram'
+
+export type RankByParam = 'total_participants' | 'total_trades' | 'total_volume'
+
+type LeaderBoardData = {
+  totalItems: number
+  rankings: ProjectRanking[]
+}
+
+type Response = {
+  code: number
+  message: string
+  data?: LeaderBoardData
+}
+
+const generateUrl = ({ id, rankBy, page = 1, pageSize = 5 }: Args) => {
+  if (!id || !rankBy) {
+    return ''
+  }
+
+  return `${SWR_KEYS.getGrantProgramLeaderBoard(id)}?${stringify({ rankBy, page, pageSize })}`
+}
+
+type Args = { id?: number; rankBy: RankByParam; page?: number; pageSize?: number }
+
+const useGetLeaderboardGrantProgram = (args: Args) => {
+  const url = generateUrl(args)
+  const swrData = useSWR<LeaderBoardData>(
+    url,
+    async (url: string) => {
+      const response = await axios.get<Response>(url)
+      if (response.data && response.data.data) {
```

Comment:
response?.data?.data

Reply:
Yeah, we don't want FE get crashed by the response.